docs/src/index.md

```diff
@@ -0,0 +1,179 @@
+# MCMCChains
+
+Implementation of Julia types for summarizing MCMC simulations and utility functions for [diagnostics](@ref Diagnostics) and [visualizations](@ref StatsPlots.jl).
+
+## Chains type
+
+```@docs
+Chains
+```
+
+## Indexing and parameter Names
+
+Chains can be constructed with parameter names.
+For example, to create a chains object with
+
+- 500 samples,
+- 2 parameters (named `:a` and `:b`)
+- 3 chains
+
+use
+
+```@example index
+using MCMCChains # hide
+using Random # hide
+Random.seed!(0) # hide
+val = rand(500, 2, 3)
+chn = Chains(val, [:a, :b])
+```
+
+By default, parameters will be given the name `:param_i`, where `i` is the parameter
+number:
+
+```@example index
+chn = Chains(rand(100, 2, 2))
+```
+
+We can set and get indexes for parameter 2:
+
+```@example index
+chn_param2 = chn[1:5,2,:];
+```
+
+```@example index
+chn[:,2,:] = repeat([4], inner = (100, 1, 2))
+chn
+```
+
+## Rename Parameters
+
+Parameter names can be changed with the function `replacenames`:
+
+```@docs
+replacenames
+```
+
+## Sections
+
+Chains parameters are sorted into sections that represent groups of parameters, see 
+[`MCMCChains.group`](@ref).
+By default, every chain contains a `:parameters` section, to which all unassigned parameters are
+assigned to. Chains can be assigned a named map during construction:
+
+```@example index
+chn = Chains(rand(100, 4, 2), [:a, :b, :c, :d])
+```
+
+The [`MCMCChains.set_section`](@ref) function returns a new `Chains` object:
+
+```@example index
+chn2 = set_section(chn, Dict(:internals => [:c, :d]))
+```
+
+Note that only `:a` and `:b` are being shown. You can explicity retrieve
+an array of the summary statistics and the quantiles of the `:internals` section by
+calling `describe(chn; sections = :internals)`, or of all variables with
+`describe(chn; sections = nothing)`. Many functions such as [`MCMCChains.summarize`](@ref) or
+[`MCMCChains.gelmandiag`](@ref) support the `sections` keyword argument.
+
+## Groups of parameters
+
+You can access the names of all parameters in a `chain` that belong to the group `:name` by using
```

Comment:
I've tried to implement the second point, but wasn't able to use `name` instead of `:name` everywhere. At some points, I was. But at other points, it lead to two problems:
1. What should be the content of variable `a` be? For example, with `a = :A`, it is still slightly more confusing for the reader when comparing the code to the output.
2. The code is harder to copy paste, because the snippets are less stand alone.

For now, I'm changing it to where possible.

Reply:
> The signature of the docstring contains only sym - we should make this consistent and probably use something like group everywhere here (also in the docstring and the signature).

I've also tried this, but found a problem here too. There is a function called `group` below `namesingroup`. If I change the parameter in `namesingroup` to `group`, then I should change the parameter of `group` to `group` too for consistency.